src/ucp/tag/rndv.c

```diff
@@ -37,24 +37,23 @@ static void ucp_rndv_rma_request_send_buffer_dereg(ucp_request_t *sreq)
      *       (state->dt.contig.memh != UCT_MEM_HANDLE_NULL)
      */
     if (UCP_DT_IS_CONTIG(sreq->send.datatype) &&
-        ucp_ep_is_rndv_lane_present(sreq->send.ep, 0) &&
-        ucp_request_is_send_buffer_reg(sreq)) {
+        ucp_ep_is_rndv_lane_present(sreq->send.ep, 0)) {
         ucp_request_send_buffer_dereg(sreq);
     }
 }
 
-size_t ucp_tag_rndv_pack_rkey(ucp_request_t *sreq, ucp_lane_index_t lane,
-                              void *rkey_buf, uint16_t *flags)
+static size_t ucp_tag_rndv_pack_rkey(ucp_request_t *sreq, void *rkey_buf, uint16_t *flags)
 {
     ucp_ep_h ep = sreq->send.ep;
+    ucp_lane_index_t lane = ucp_ep_get_rndv_get_lane(ep, 0);
     ucs_status_t status;
 
     ucs_assert(UCP_DT_IS_CONTIG(sreq->send.datatype));
 
     /* Check if the sender needs to register the send buffer -
      * is its datatype contiguous and does the receive side need it */
     if (ucp_ep_rndv_md_flags(ep, 0) & UCT_MD_FLAG_NEED_RKEY) {
```

Comment:
I guess you need to pass `lane` instead of `0`

Reply:
right, it seems this is long-play error :)